kge/job/auto_search.py

```diff
@@ -91,7 +91,7 @@ def get_best_parameters(self):
 
     # -- Main --------------------------------------------------------------------------
 
-    def _run(self):
+    def _run(self, job_trace : Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Neither would I add the job_trace argument here (it can be a member of the job, if really needed) nor should the run method be required to return a Trace (at the Job level, specific jobs may do this). So, `Job.run` should have signature run() -> Any. Likewise the corresponding post_run hook. Together with the job itself, that's all the hooks will ever need. This way, we keep things simple and extensible.

Reply:
TrainJob and SerachJob can make the return type more specific, of course.